[REFACTOR] Board API 리팩토링 및 수정

tnals2384 opened the pull request:
### PR 타입(하나 이상의 PR 타입을 선택해주세요)
-[x] 기능 추가
-[] 기능 삭제
-[] 버그 수정
-[] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치
ex) refactor/#95/board-refactoring -> develop

### 변경 사항

 1. response **boolean -> Boolean 타입 변경**( isLiked, isFixed)
---
 2. 댓글 list 반환 시 **본인 댓글**인지 여부 반환 (isMine)
 3. 상세 조회 시 **본인 글**인지 여부 반환 (isMine)
 ---
 4. **대댓글** 작성 기능
  - 대댓글 기능 추가를 위해 commentId 파라미터 추가 -> 대댓글인 경우만 사용
  - 대댓글 삭제 시 하위 댓글 존재 여부에 따라 삭제 처리
  - 댓글 list 반환 시 해당 댓글의 대댓글 수 반환(childrenNumber)
  - 해당 댓글의 대댓글 조회 API 추가
 ---
 5.  DataConverter 사용하여 createdAt, writer 반환
 ---
 6.  **WriterInfo DTO**를 생성하여 Board게시판에 필요한 writer, profileImage, Semester, Part 반환
   - 글 목록, 검색 목록, 핀고정 공지 목록 조회 시에는 writer, profileImage만 반환
   - 글 상세 조회, 댓글 조회 시에는 writer, profileImage, Semester, Part 반환
   - 내 글 목록, 운영진용 공지 목록 조회 시에는 writer만 필요하므로, 해당 DTO 사용하지 않음 
 7. **BoardPageInfos DTO** 를 통해 중복되는 PageInfos 코드 제거
 ---
 8. **핀 고정된 공지(최대 10개) 조회 API**
   -  hostType null로 줄 경우 모든 host의 핀공지, hostType을 줄 경우 해당 hostType의 핀공지를 반환합니다.
   - app 뷰에 맞춰 response에 사용자 정보, heartCount, commentCount, thumbnail 추가
 9. **글 목록 조회 시 고정된 게시글은 조회하지 않도록** 수정 ( 핀 고정 공지 API가 따로있기 때문에 )
 10. 운영진용 공지 조회 목록 조회 API 수정
  -- hostType All enum을 삭제하고, hostType null일 경우 운영진이 핀 고정할 수 있는 모든 공지 조회, hostType이 주어지면 해당 hostType의 공지를 조회

### 테스트 결과
<img width="590" alt="스크린샷 2024-03-27 오전 1 26 25" src="https://github.com/Team-UMC/UMC-Server/assets/83461362/fb0c0710-0ec9-44e9-8142-c71a2d794d03">
<img width="565" alt="스크린샷 2024-03-27 오전 1 25 20" src="https://github.com/Team-UMC/UMC-Server/assets/83461362/baaee6ea-053f-4c94-9be4-a09e9e445ae2">

tnals2384 committed:
fix: response dto boolean -> Boolean 수정

tnals2384 committed:
fix: board 생성 시 기본 핀고정 false

tnals2384 committed:
fix: pinnedNotice response dto  필드 수정
- writer, createdAt, hitCount 필드 추가

tnals2384 committed:
fix: 게시글 상세 조회 response에 isMine 추가

tnals2384 committed:
style: 메서드 주석 작성

tnals2384 committed:
fix: comment 목록 조회 response에 isMine 추가

tnals2384 committed:
fix: 불필요한 member 파라미터 제거

tnals2384 committed:
fix: loadEntity로 member Entity 불러와서 로직 수행

tnals2384 committed:
feat: 대댓글 작성 API

tnals2384 committed:
test: 대댓글 작성 API 서비스 테스트 코드 작성

tnals2384 committed:
fix: 대댓글 작성 컨트롤러,서비스 삭제 -> 댓글 작성 API에서 해결

tnals2384 committed:
test: 대댓글 작성 서비스 테스트코드 수정, 컨트롤러 테스트코드 생성

tnals2384 committed:
resolved merge conflict

tnals2384 committed:
refactor: boardPageInfos 코드 분리 후 재사용

tnals2384 committed:
fix: 핀고정된 공지사항 최대 10개까지 조회, isPinned -> isFixed 파라미터명 변경

tnals2384 committed:
feat: DataConverter를 이용하여 createdAt 반환

tnals2384 committed:
feat: DataConverter를 이용하여 writer 반환

tnals2384 committed:
refactor: 부분 코드 리팩토링

tnals2384 committed:
refactor: writerInfo를 통해 writer의 정보를 반환

tnals2384 committed:
refactor: 게시글 목록에서도 WriterInfo를 사용하도록 수정

tnals2384 committed:
fix: 핀고정 공지 조회 api response 수정
- app 뷰에 맞춰 response에 사용자 정보, heartCount, commentCount, thumbnail 추가
- 핀고정 공지를 따로 반환하기 때문에, 게시글 목록 조회 api 응답에서도 isFixed 필드 삭제
-

tnals2384 committed:
fix: hostType에 따라 핀고정 공지 조회할 수 있도록 api 수정

tnals2384 committed:
fix: staff 공지사항 조회 API 수정
- hostType에 ALL 삭제, 파라미터로 hostType을 주지 않을 경우(null일 경우) 핀고정 권한이 있는 모든 host의 글을 조회하도록 수정

tnals2384 committed:
test: board service, controller 테스트 코드 수정

tnals2384 committed:
test: board service, controller 테스트 코드 수정

tnals2384 committed:
style: 간단한 코드 정리

tnals2384 committed:
fix: 게시글 목록 조회 시 핀고정된 게시글은 조회하지 않음

tnals2384 committed:
fix: 댓글 삭제 로직 변경, 함수명 변경
- 댓글 삭제시 대댓글이 있으면 댓글 내용을 변경, 없으면 delete

tnals2384 committed:
test : 댓글 삭제 test code 추가

junseokkim commented on src/main/java/com/umc/networkingService/domain/board/service/BoardCommentServiceImpl.java:
```diff
@@ -58,43 +76,63 @@ public BoardCommentResponse.BoardCommentId updateBoardComment(Member member, UUI
+    public BoardCommentPageInfos<BoardCommentPageElement> showBoardComments(Member loginMember, UUID boardId, Pageable pageable) {
+
+        Member member = memberService.loadEntity(loginMember.getId());
         Board board = boardService.loadEntity(boardId);
```
Board에 속한 댓글을 조회하기 전에, 해당 Board를 볼 수 있는 권한이 있는지 확인하는 부분(예를 들어, 본인 학교나 지부 글이 맞는지) 있으면 좋을 거 같아요

tnals2384 replied:
수정하겠습니다!!

junseokkim commented on src/main/java/com/umc/networkingService/domain/board/entity/BoardComment.java:
```diff
@@ -44,4 +44,11 @@ public void update(BoardCommentRequest.BoardCommentUpdateRequest request) {
         this.content = request.getContent();
     }
 
+    public void setParentComment(BoardComment parentComment) {
```
해당 함수는 사용이 안되는데, 추후에 사용을 위한 건가요???

tnals2384 replied:
삭제하겠습니다!!

tnals2384 committed:
feat: 게시판 읽기 권한 check 함수 추가

tnals2384 committed:
test: 게시판 읽기 권한 check 함수 test

tnals2384 committed:
resolved merge conflict

tnals2384 committed:
feat: 댓글 조회 API 수정

tnals2384 committed:
feat: 대댓글 조회 API 추가 / 기존 댓글 조회 API 수정
- 대댓글을 조회하고자 하는 댓글의 commentId를 파라미터로 전달하면 대댓글 조회 API를 통해 조회
- 기존 댓글 조회 API에서는 해당 댓글의 대댓글 수만 반환

tnals2384 committed:
feat: 대댓글 조회 API 추가 / 기존 댓글 조회 API 수정
- 대댓글을 조회하고자 하는 댓글의 commentId를 파라미터로 전달하면 대댓글 조회 API를 통해 조회
- 기존 댓글 조회 API에서는 해당 댓글의 대댓글 수만 반환

junseokkim reviewed (approved):
LGTM

tnals2384 committed:
지운코드 복원

tnals2384 committed:
fix: url 수정
